roles/nodejs/tasks/main.yml

```diff
@@ -41,6 +41,12 @@
 #      /etc/iiab/local_vars.yml -- then re-run this IIAB installer.
 #  when: nodejs_version_installed is defined and nodejs_version_installed.stdout != nodejs_version and nodejs_version_installed.stderr == ""
 
+- name: Remove /etc/apt/sources.list.d/nodesource.list if nodejs_version_installed.stdout is less than {{ nodejs_version }}
+  file:
+    state: absent
+    path: /etc/apt/sources.list.d/nodesource.list
+  when: nodejs_version_installed is defined and nodejs_version_installed.stdout != nodejs_version and nodejs_version_installed.stdout != ""
```

Comment:
Per the description 4 lines above, let's actually test to see if the installed Node.js version is **less than** desired (i.e. too old rather than too new) just like we use Ansible to determine if "PHP version is too old" for some versions of Nextcloud here: (https://github.com/iiab/iiab/blob/master/roles/nextcloud/tasks/install.yml#L34)

```
- name: Set var 'php_new' indicating if installed version of PHP ({{ php_version_installed.stdout }}) >= 7.1, as required by Nextcloud 16
  set_fact:
    php_new: "{{ php_version_installed.stdout is version('7.1', '>=') }}"
    # Ansible's Version Comparison routine:
    # https://docs.ansible.com/ansible/latest/user_guide/playbooks_tests.html#version-comparison
  when: php_version_installed.stdout != ""    # i.e. IF ABOVE 'php -v' WORKED
```

Reply:
I just reused the exact same logic that the stanza below uses to uninstall nodejs when the version is not correct. If you wish to use an integer comparison please alter the nodejs_version_installed stanza to suit your taste, change this stanza and the one below please.